### PR TITLE
Feature/grid table rename and advanced grid

### PR DESCRIPTION
**Pull request by romaninsh**
I have decided to split this change up into multiple smaller chunks. 

The first one renames "Grid" into "Table" and also "Column" namespace into "TableColumn". This is needed because I'm going to add class "Column" which would otherwise clash with table columns. This is not backwards-compatible change, as we are still early on in the versions and creating backwards compatibility would be too messy.

### CODE REVIEW

File: src/TableColumn/Actions.php [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+/**
+ * Formatting money.
+ */
+class Actions extends Generic
+{
+    public $actions = [];
+
+    public function addAction($icon, $callback)
+    {
+        $cb = $this->add(new jsCallback(), $action);
```
Comment (DarkSide666) [inline src/TableColumn/Actions.php]: There is no variable $action and not used parameter $icon. I guess parameter should be named - $action.

File: src/TableColumn/Link.php [hunk 2]
```diff
@@ -1,14 +1,25 @@
...
 class Link extends Generic
 {
+    /**
+     * TODO: Refactor so that this goes away!!
+     */
     public $page = null;
 
+    /**
+     * Use
+     *   new Link('http://yahoo.com?id={$id}');
```
Comment (DarkSide666) [inline src/TableColumn/Link.php]: This syntax is not supported as I see below in constructor

File: src/TableColumn/Link.php [hunk 3]
```diff
@@ -36,6 +47,14 @@ public function getCellTemplate(\atk4\data\Field $f)
         return $output;
     }
 
+    /**
+     * Auto-fill tag for the row-id.
+     *
+     * @param [type] $row   [description]
```
Comment (DarkSide666) [inline src/TableColumn/Link.php]: need to set correct parameter types

File: src/TableColumn/Actions.php (hunk 1 above)
Comment (romaninsh) [inline src/TableColumn/Actions.php]: oh actually Action is incomplete., i should have dropped it.

Comment (romaninsh) [inline src/TableColumn/Actions.php]: addressed (removing Actions for now)

**Pull request merged**